internal/fsutil/os_detection.go

```diff
@@ -0,0 +1,55 @@
+package fsutil
+
+import (
+	"os"
+	"strings"
+)
+
+type OSInfo struct {
+	Name    string
+	Version string
+}
+
+type Detector interface {
```

Comment:
iirc this breaks one of those golang dictates of like, "define the interface where you need it, not alongside the first struct that implements it" but on the other hand it's not like the golang core team is going to personally slap me in the face for this infraction.   lmk if we have standards i should follow...

Reply:
Why do we need the interface? It looks like it only has one implementation. If we wanted we could make a no-op implementation for not-linux, then `HasLinuxFile()` could go away.